src/cephadm/cephadm

```diff
@@ -4931,7 +4931,7 @@ def infer_mon_network(ctx: CephadmContext, mon_eps: List[EndPoint]) -> Optional[
     mon_networks = []
```

Comment:
run-tox-cephadm fails with:
```
mypy: commands[0] /home/jenkins-build/build/workspace/ceph-pull-requests/src/cephadm> mypy --config-file ../mypy.ini cephadm
cephadm: note: In class "ContainerEngine":
cephadm:233: error: Only instance methods can be decorated with @property
Found 1 error in 1 file (checked 1 source file)
mypy: exit 1 (7.08 seconds) /home/jenkins-build/build/workspace/ceph-pull-requests/src/cephadm> mypy --config-file ../mypy.ini cephadm pid=3572485
mypy: FAIL ✖ in 14.42 seconds
```

Not sure if this can also be fixed in a backport or not.

Reply:
This is because https://github.com/ceph/ceph/commit/7440d7a3c07d3e83ebb2ea7ac9cbbd94762e78ed was never backported I think. We could probably just pull that commit in here as part of the backport as what it is doing is how we'd want to fix this anyway. In fact, maybe we should just include the entirety of https://github.com/ceph/ceph/pull/48557 in this backport. If we're backporting fixes to get things to work with a higher version of mypy, including things that were meant to help the tox suite run with higher python versions makes sense.

Reply:
I have included those , pls check @adk3798 

Thanks!